File: go/0383E-赎金信.go
/*
 * @Author: SukiEva
 * @Date: 2021-10-16 19:37:51
 * @Description: https://github.com/SukiEva
 * わたし、気になります！
 */

package main

func canConstruct(ransomNote string, magazine string) bool {
	m := [30]int{}
	for _, v := range magazine {
		m[v-'a']++
	}
	for _, v := range ransomNote {
		m[v-'a']--
		if m[v-'a'] < 0 {
			return false
		}
	}
	return true
}

File: cpp/0383E-赎金信.cpp
/*
 * @Author: SukiEva
 * @Date: 2021-12-15 08:44:58
 * @Description: https://github.com/SukiEva
 * わたし、気になります！
 */
#include <bits/stdc++.h>

#include "include/ListNode.h"
#include "include/TreeNode.h"

using namespace std;

class Solution {
   public:
    bool canConstruct(string ransomNote, string magazine) {
        int vis[27] = {0};
        for (auto c : magazine) {
            vis[c - 'a']++;
        }
        for (auto c : ransomNote) {
            vis[c - 'a']--;
            if (vis[c - 'a'] < 0) {
                return false;
            }
        }
        return true;
    }
};